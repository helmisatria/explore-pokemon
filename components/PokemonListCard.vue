<template>
  <article
    class="`pt-5 pb-3 relative rounded-xl shadow flex flex-col items-center transition ease ring-2 ring-transparent hover:ring-gray-400 hover:ring-offset-2 `"
    :class="[`${backgroundColor} bg-opacity-30`]"
  >
    <div
      class="absolute leading-none top-[2px] left-[2px] rounded-tl-xl rounded py-1 px-2 flex items-center"
      :class="`${backgroundColor} bg-opacity-40`"
    >
      <span class="text-xs text-white">#{{ formatId(String(pokemon.id)) }}</span>
    </div>
    <img
      :src="`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/${pokemon.id}.png`"
      :alt="pokemon.name"
      loading="lazy"
      height="144"
      width="144"
      class="w-[9rem] h-auto"
    />
    <h2 class="mt-2 text-center capitalize font-semibold">
      {{ pokemon.name }}
    </h2>

    <PokemonLabelTypes :types="pokemon.pokemons[0].types" class="mt-2" variation="small" />
  </article>
</template>

<script>
import { formatId } from '../utils/string-helpers'
import PokemonLabelTypes from './PokemonLabelTypes.vue'

export default {
  name: 'PokemonListCard',
  components: { PokemonLabelTypes },
  props: {
    pokemon: {
      type: Object,
      default: () => ({
        id: null,
        name: null,
      }),
    },
  },
  computed: {
    backgroundColor() {
      return `bg-pokemon-${this.pokemon.pokemons[0].types[0].type.name}`
    },
  },
  methods: {
    formatId,
  },
}
</script>
