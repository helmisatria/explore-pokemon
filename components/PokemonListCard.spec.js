import { createLocalVue, shallowMount } from '@vue/test-utils'
import { PiniaVuePlugin } from 'pinia'
import { createTestingPinia } from '@pinia/testing'
import PokemonListCard from './PokemonListCard.vue'

import pokemonsData from '~/fixtures/pokemons.json'

describe('Component: PokemonListCard', () => {
  let wrapper

  const localVue = createLocalVue()
  localVue.use(PiniaVuePlugin)

  const generateWrapper = () => {
    return shallowMount(PokemonListCard, {
      localVue,
      pinia: createTestingPinia(),
      propsData: {
        pokemon: pokemonsData.species[0],
      },
    })
  }

  beforeEach(() => {
    wrapper = generateWrapper()
  })

  it('should mount component properly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
