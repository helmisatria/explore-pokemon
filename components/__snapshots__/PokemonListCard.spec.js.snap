// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PokemonListCard should mount component properly 1`] = `
<article
  class="\`pt-5 pb-3 relative rounded-xl shadow flex flex-col items-center transition ease ring-2 ring-transparent hover:ring-gray-400 hover:ring-offset-2 \` bg-pokemon-grass bg-opacity-30"
>
  <div
    class="absolute leading-none top-[2px] left-[2px] rounded-tl-xl rounded py-1 px-2 flex items-center bg-pokemon-grass bg-opacity-40"
  >
    <span
      class="text-xs text-white"
    >
      #001
    </span>
  </div>
   
  <img
    alt="bulbasaur"
    class="w-[9rem] h-auto"
    height="144"
    loading="lazy"
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png"
    width="144"
  />
   
  <h2
    class="mt-2 text-center capitalize font-semibold"
  >
    
    bulbasaur
  
  </h2>
   
  <pokemonlabeltypes-stub
    class="mt-2"
    types="[object Object],[object Object]"
    variation="small"
  />
</article>
`;
